decode_unmarshalable_string_test.go

```diff
@@ -0,0 +1,61 @@
+package yaml
```

Comment:
Would you please add test case in `decode_test.go` ?

Reply:
done.